pkg/options/describe_test.go

```diff
@@ -0,0 +1,286 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package options
+
+import (
+	"testing"
+
+	"github.com/AlecAivazis/survey/v2/terminal"
+	goexpect "github.com/Netflix/go-expect"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/test"
+	"github.com/tektoncd/cli/test/prompt"
+)
+
+func TestDescribeOptions_ValidateOpts(t *testing.T) {
+
+	testParams := []struct {
+		name      string
+		limit     int
+		wantError bool
+		want      string
+	}{
+		{
+			name:      "valid limit",
+			limit:     1,
+			wantError: false,
+			want:      "",
+		},
+		{
+			name:      "invalid limit",
+			limit:     0,
+			wantError: true,
+			want:      "limit was 0 but must be a positive number",
+		},
+	}
+
+	for _, tp := range testParams {
+		t.Run(tp.name, func(t *testing.T) {
+			opts := NewDescribeOptions(&cli.TektonParams{})
+			opts.Limit = tp.limit
+
+			err := opts.ValidateOpts()
+			if tp.wantError {
+				if err == nil {
+					t.Errorf("Error expected here")
+				}
+				test.AssertOutput(t, tp.want, err.Error())
+			} else if err != nil {
+				t.Errorf("unexpected Error")
+			}
+		})
+	}
+}
+
+func TestDescribeOptions_Ask(t *testing.T) {
+
+	options := []string{
+		"pipeline1",
+		"pipeline2",
+		"pipeline3",
+	}
+
+	options2 := []string{
+		"sample-pipeline-run1 started 1 minutes ago",
+		"sample-pipeline-run2 started 2 minutes ago",
+		"sample-pipeline-run3 started 3 minutes ago",
+	}
+
+	options3 := []string{
+		"task1",
+		"task2",
+		"task3",
+	}
+
+	options4 := []string{
+		"sample-task-run1 started 1 minutes ago",
+		"sample-task-run2 started 2 minutes ago",
+		"sample-task-run3 started 3 minutes ago",
+	}
+
+	options5 := []string{
+		"pipeline-resource1",
+		"pipeline-resource2",
+		"pipeline-resource3",
+	}
+	options6 := []string{
+		"clustertask1",
+		"clustertask2",
+		"clustertask3",
+	}
+
+	testParams := []struct {
+		name     string
+		resource string
+		prompt   prompt.Prompt
+		options  []string
+		want     DescribeOptions
+	}{
+		{
```

Comment:
All of the tests here seem to only be testing selecting the first object. Please add more tests to select a different object than just the first in the list.